### PR TITLE
Enable code coverage, spot bugs, and check styles

### PR DESCRIPTION
**Pull request by BuddhiWathsala**
## Purpose
$subject
Subtask of https://github.com/ballerina-platform/ballerina-standard-library/issues/852

## Test environment
- macOS
- SLAlpha3
- Java11

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -23,7 +23,7 @@ plugins {
 }
 
 ext.ballerinaLangVersion = project.ballerinaLangVersion
-ext.puppycrawlCheckstyleVersion = "8.18"
+ext.puppycrawlCheckstyleVersion = project.puppycrawlCheckstyleVersion
 ext.testngVersion = "6.14.3"
 ext.slf4jVersion = "1.7.30"
```
Comment (ThisaruGuruge) [inline build.gradle:28]: If we can move these two to the `gradle.properties` file as well, we can remove all these lines with `ext.<name>`. They'll be picked up from the gradle.properties file.

Comment (BuddhiWathsala) [reply build.gradle]: Fixed with https://github.com/ballerina-platform/module-ballerina-reflect/pull/47/commits/2a892ab9d7e79471e0bf777eb282a8324637f3ea

File: build.gradle [hunk 2]
```diff
@@ -23,9 +23,8 @@ plugins {
...
 
 ext.ballerinaLangVersion = project.ballerinaLangVersion
-ext.puppycrawlCheckstyleVersion = "8.18"
-ext.testngVersion = "6.14.3"
-ext.slf4jVersion = "1.7.30"
+ext.puppycrawlCheckstyleVersion = project.puppycrawlCheckstyleVersion
+ext.checkstyleToolVersion = project.checkstyleToolVersion
 ext.stdlibRegexVersion = project.stdlibRegexVersion
 ext.stdlibOsVersion = project.stdlibOsVersion
 ext.stdlibIoVersion = project.stdlibIoVersion
```
Comment (ThisaruGuruge) [inline build.gradle:31]:
```suggestion
```

Comment (BuddhiWathsala) [reply build.gradle]: Fixed in https://github.com/ballerina-platform/module-ballerina-reflect/pull/47/commits/8f05ed77dca94e9d0268634e27c526e97151ece4

**Review by ThisaruGuruge — APPROVED** [review]

**Pull request merged**